Lib9c/Action/ItemEnhancement.cs

```diff
@@ -280,46 +319,40 @@ public override IAccountStateDelta Execute(IActionContext context)
             return states.SetState(slotAddress, slotState.Serialize());
         }
 
-        protected override IImmutableDictionary<string, IValue> PlainValueInternal
+        public static EnhancementResult GetEnhancementResult(EnhancementCostSheetV2.Row row, IRandom random)
         {
-            get
+            var greatSuccessRatio = (int)(row.GreatSuccessRatio * GameConfig.TenThousand);
+            var successRatio = (int)(row.SuccessRatio * GameConfig.TenThousand) + 1;
```

Comment:
Is there any reason to calculate ratio instead of set the row?

Reply:
The row is up to the second decimal place, to see the probability to the fourth decimal place.